### PR TITLE
Implement the widget toolkit brick

### LINKED ISSUE #417: Build a brick for the UI toolkit
ACs

The same components implemented in the [widget_toolkit example](https://github.com/Prime-Holding/widget_toolkit/issues/2)

Notes:
- Add a page to showcase all Widgets
- Add Route in the Router and a button to navigate to the example page
- Add required dependencies into pubspec.yaml
- Document all additional settings that the developer needs to do if the brick is not part of the initial project. Those are adding dependencies to pubspeck,yaml, Route, additional app level dependencies, buttons to open example pages e.t.

### PR DESCRIPTION
**Pull request by StanevPrime**
Closes #417

### CODE REVIEW

File: packages/rx_bloc_cli/bin/compile.sh [hunk 1]
```diff
@@ -4,11 +4,23 @@ dart run mason_cli:mason bundle \
   -t dart \
   -o lib/src/templates/ \
   mason_templates/bricks/rx_bloc_base
+
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/bin/compile.sh:7]: @JvnSlv, please apply the same commands for the win version of the file

File: packages/rx_bloc_cli/lib/src/commands/create_command.dart [hunk 2]
```diff
@@ -227,6 +242,12 @@ class CreateCommand extends Command<int> {
     return counterEnabled.toLowerCase() == 'true';
   }
 
+  /// Returns whether the project will be created with widget toolkit feature
+  bool _parseEnableWidgetToolkit(ArgResults arguments) {
+    final counterEnabled = arguments[_widgetToolkitString];
```
Comment (JvnSlv) [inline packages/rx_bloc_cli/lib/src/commands/create_command.dart:247]: This can be renamed to the more suitable name

**Review by JvnSlv — APPROVED** [review]

File: packages/rx_bloc_cli/lib/src/commands/create_command.dart [hunk 3]
```diff
@@ -308,7 +329,9 @@ class CreateCommand extends Command<int> {
 
     _usingLog('Firebase Analytics', arguments.enableAnalytics);
     _usingLog('Firebase Push Notifications', true);
-    _usingLog('Feature Counter', arguments.enableCounterFeature);
+    _usingLog('Feature Counter Showcase', arguments.enableCounterFeature);
+    _usingLog('Firebase Widget Toolkit Showcase',
```
[the remark this answers is not in the document]
Comment (StanevPrime) [reply packages/rx_bloc_cli/lib/src/commands/create_command.dart:333]: Done

**Review by RomanovaPrime — COMMENTED** [review]
Maybe edit or remove if not needed README and CHANGELOG files for newly added bricks.

**Pull request merged**